File: lib/result/Ring.java
package com.teamlab.communication.wireless.lpr9201.result;

/**
 * リザルト Ring
 */
public class Ring extends Result {
	/**
	 * 
	 */
	public static final int RESULT_CODE = 0x87;

	/**
	 * 
	 */
	public static final int DATA_LENGTH_SIZE = 1; // byte

	/**
	 * 
	 * @param datas
	 */
	public Ring(int[] datas) {
		super(RESULT_CODE, DATA_LENGTH_SIZE, datas);
	}
}

File: lib/result/ring.js
/* jslint node: true */
"use strict"

/**
 * リザルト Ring
 */

exports.Ring = Ring;

/**
 * 結果コード
 */
Ring.RESULT_CODE = 0x87;

/**
 * データ長バイトサイズ
 */
Ring.DATA_LENGTH_BYTE_SIZE = 1; // byte

/**
 * @param {Result} result
 */
function Ring(result) {
  this.result = result;
}

/**
 * パース可能か
 * @param {Result} result
 */
Ring.canParse = function(result) {
  return result !== null && result.resultCode === Ring.RESULT_CODE;
}
